### PR TITLE
Django support sensitive variables

### LINKED ISSUE #2356: [DJANGO] Support sensitive_variables decorators in integration
### Problem Statement

In the Django framework, a utility exists to declare that a variable is sensitive or not.

https://docs.djangoproject.com/en/3.2/_modules/django/views/decorators/debug/

Would it be possible to support this functionality in the integration directly?

### Solution Brainstorm

The SDK tries to read the sensitive_variables attribute on the functions it passes to add them to sensitive information not to be displayed clearly in sentry.

### PR DESCRIPTION
**Pull request by vagi8**
Fix #2356 

- Added support for sensitive_variables and sensitive_post_parameter decorators.
- Added test cases

### CODE REVIEW

**Comment by vagi8** [comment]
Future Task: Django version 5.0 is expected to support coroutine functions as well. This PR does not add support for it. I will deal with it in a separate PR.

**Review by szokeasaurusrex — COMMENTED** [review]
@vagi8 Thank you for your contribution!

It appears that some of the Django tests you wrote are currently failing. The output from our CI test run is available [here](https://github.com/getsentry/sentry-python/actions/runs/6349471127/job/17247753798); if you can see it, that's great, but it is possible that it is only available to Sentry employees. If that's the case, let me know, and I can send you the output from the failing tests directly.

You can also run the Django tests locally using the following command: `tox -e py3.11-django-v4.1` (this requires Tox to be installed and setup correctly, please let me know if you need guidance on how to do this).

Once you've fixed the failing tests, we will be happy to take a more thorough look at this PR. Please also feel free to reach out to us if you need help setting up the local test environment or any assistance with debugging.

**Comment by sentrivana** [comment]
Hey @vagi8. Could you try removing/commenting out this test https://github.com/getsentry/sentry-python/blob/master/tests/integrations/django/test_basic.py#L1057 and seeing if the problem persists? I found out in an unrelated PR that that test may be altering the urlconf for other tests, leading to failures. I'll see if I can fix that in another PR.

**Comment by vagi8** [comment]
@sentrivana  
All test cases pass when I comment out the said test case. Verified with django-v2.2 and django-v4.1

**Comment by sentrivana** [comment]
@vagi8 Thanks for verifying! I've pushed a small commit to your PR to address the broken test case.

**Comment by sentrivana** [comment]
@vagi8 Can you please take a look at these?

- the Django Python 2.7 tests are failing: https://github.com/getsentry/sentry-python/actions/runs/6638956405/job/18036359266?pr=2399
- mypy complaining in the lint step: https://github.com/getsentry/sentry-python/actions/runs/6638956425/job/18036360400?pr=2399

**Comment by vagi8** [comment]
The past few hours I have been trying to setup the py2.7 environment locally and it is a pain. Not sure if i did setup correctly, when I run pytest on django i get this error and  cant seem to resolve it. 

> tests/integrations/django/test_basic.py:28: in <module>
    from sentry_sdk.integrations.executing import ExecutingIntegration
sentry_sdk/integrations/executing.py:17: in <module>
    raise DidNotEnable("executing is not installed")
E   DidNotEnable: executing is not installed

`tox -e py2.7-django-v1.8` has similar error.

File: sentry_sdk/integrations/django/__init__.py [hunk 1]
```diff
@@ -539,6 +648,25 @@ def raw_data(self):
...
+                # Cleanse all parameters.
+                for k in cleansed_request:
+                    cleansed_request[k] = SENSITIVE_DATA_SUBSTITUTE
+            else:
+                # Cleanse only the specified parameters.
+                for param in sensitive_post_parameters:
+                    if param in cleansed_request:
+                        cleansed_request[param] = SENSITIVE_DATA_SUBSTITUTE
+            return cleansed_request
         return self.request.POST
```
Comment (szokeasaurusrex) [inline sentry_sdk/integrations/django/__init__.py:670]: We should try to rewrite this function more concisely; below is one way we might do that:
```suggestion
        # type: () -> dict[str, Any]
        """
        Clean form data of any sensitive post parameters defined in the django app
        see https://docs.djangoproject.com/en/3.2/_modules/django/views/decorators/debug/
        """
        sensitive_post_parameters = getattr(
            self.request, "sensitive_post_parameters", []
        )

        if sensitive_post_parameters == "__ALL__":
            sensitive_post_parameters = self.request.POST.keys()

        return {
            k: SENSITIVE_DATA_SUBSTITUTE if k in sensitive_post_parameters else v
            for k, v in self.request.POST.items()
        }
```

**Review by szokeasaurusrex — COMMENTED** [review]
Hey @vagi8, I have been pretty busy for the past few weeks, but I finally had some time to review your PR and add my feedback. Please let me know if you have any questions about my feedback!

**Comment by vagi8** [comment]
Hey @szokeasaurusrex, Thank you for being patient on the PR. Thanks for the feedback. I resolved everything.

**Comment by antonpirker** [comment]
How are we on this @vagi8 ? Is there some time or motivation to bring this over the finish line?

**Comment by vagi8** [comment]
@antonpirker little busy but I can try to spend some time. Any deadline ? We Django 5 released as well so I might have to add support for coroutines too.

**Comment by antonpirker** [comment]
We do not have deadlines :-)

**Comment by sentrivana** [comment]
I don't know what's up with stalebot, closing manually. Please open a new PR if you find some time to continue on this @vagi8 🙏🏻

**Pull request closed without merging**